web/modules/custom_weather_module/custom_weather_module.install

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @file
+ * Checks if custom_weather_module.settings available.
+ */
+
+/**
+ * Implements hook_install().
+ */
+function custom_weather_module_install() {
+  // @todo use dependency injection
+  // phpcs:ignore
+  if (!\Drupal::config('custom_weather_module.settings')->get('api_key')) {
```

Comment:
the ->get('api_key') will always return a config. Did you test it with a non-existed key?